data_steward/tools/create_combined_backup_dataset.py

```diff
@@ -0,0 +1,540 @@
+
+    :param client: a BigQueryClient
+    :param rdr_dataset_id: Identifies the name of rdr dataset
+    :param combined_dataset: Identifies the name of combined dataset
+
+    :return:
+    """
+    q = combine_consts.EHR_CONSENT_QUERY.format(
+        dataset_id=rdr_dataset_id,
+        source_value_ehr_consent=combine_consts.SOURCE_VALUE_EHR_CONSENT,
+        concept_id_consent_permission_yes=combine_consts.
+        CONCEPT_ID_CONSENT_PERMISSION_YES)
+    fq_table_name = f'{client.project}.{combined_dataset}.{combine_consts.EHR_CONSENT_TABLE_ID}'
+    table = bigquery.Table(fq_table_name)
+    table = client.create_table(table, exists_ok=True)
+    LOGGER.info(f'Query for {combine_consts.EHR_CONSENT_TABLE_ID} is {q}')
+    query(client,
+          q,
+          dst_dataset_id=combined_dataset,
+          dst_table_id=combine_consts.EHR_CONSENT_TABLE_ID,
+          write_disposition=WRITE_APPEND)
+
+
+def copy_ehr_table(client: BigQueryClient, table: str, unioned_ehr_dataset: str,
+                   combined_dataset: str):
+    """
+    Copy table from EHR (consenting participants only) to the combined dataset without regenerating ids
+
+    Note: Overwrites if a table already exists
+    """
+    fields = resources.fields_for(table)
+    field_names = [field['name'] for field in fields]
+    if 'person_id' not in field_names:
+        raise RuntimeError(
+            f'Cannot copy EHR table {table}. It is missing columns needed for consent filter'
+        )
+    q = combine_consts.COPY_EHR_QUERY.format(
+        ehr_dataset_id=unioned_ehr_dataset,
+        table=table,
+        ehr_consent_table_id=combine_consts.EHR_CONSENT_TABLE_ID,
+        combined_dataset_id=combined_dataset)
+    LOGGER.info(f'Query for {table} is `{q}`')
+    query(client, q, combined_dataset, table, WRITE_APPEND)
+
+
+def mapping_query(domain_table: str, rdr_dataset: str, unioned_ehr_dataset: str,
+                  combined_dataset: str):
+    """
+    Returns query used to get mapping of all records from RDR combined with EHR records of consented participants
+
+    :param domain_table: one of the domain tables (e.g. 'visit_occurrence', 'condition_occurrence')
+    :return:
+    """
+
+    if combine_consts.PERSON_ID in [
+            field['name'] for field in resources.fields_for(domain_table)
+    ]:
+        return combine_consts.MAPPING_QUERY_WITH_PERSON_CHECK.format(
+            rdr_dataset_id=rdr_dataset,
+            ehr_dataset_id=unioned_ehr_dataset,
+            combined_dataset_id=combined_dataset,
+            domain_table=domain_table,
+            mapping_constant=common.RDR_ID_CONSTANT,
+            ehr_consent_table_id=combine_consts.EHR_CONSENT_TABLE_ID)
+    return combine_consts.MAPPING_QUERY_WITHOUT_PERSON_CHECK.format(
+        rdr_dataset_id=rdr_dataset,
+        ehr_dataset_id=unioned_ehr_dataset,
+        combined_dataset_id=combined_dataset,
+        domain_table=domain_table,
+        mapping_constant=common.RDR_ID_CONSTANT)
```

Comment:
if this were done with a Jinja template, the comparison to add the `WHERE` clause could be kept in the SQL and we wouldn't have to duplicate the remainder of the query to include a single WHERE clause.  

Thoughts?